hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -858,26 +871,37 @@ private void addToObjectIdMap(Table<String, ? extends WithObjectID> fsTable,
                  : sstFileReader.getKeyStream()) {
       keysToCheck.forEach(key -> {
         try {
-          final WithObjectID oldKey = fsTable.get(key);
-          final WithObjectID newKey = tsTable.get(key);
+          final WithParentObjectId oldKey = fsTable.get(key);
+          final WithParentObjectId newKey = tsTable.get(key);
           if (areKeysEqual(oldKey, newKey) || !isKeyInBucket(key, tablePrefixes,
               fsTable.getName())) {
             // We don't have to do anything.
             return;
           }
           if (oldKey != null) {
             byte[] rawObjId = codecRegistry.asRawData(oldKey.getObjectID());
+            // Removing volume bucket info by removing the table bucket Prefix
+            // from the key.
+            // For FSO buckets will be left with the parent id/keyname.
+            // For OBS buckets will be left with the complete path
             byte[] rawValue = codecRegistry.asRawData(
-                getKeyOrDirectoryName(isDirectoryTable, oldKey));
+                key.substring(tablePrefix.length()));
```

Comment:
You didn't do the same change for `newKey != null` case at line 899. Is it intentional or just missed?

Reply:
Yeah I missed it.